### PR TITLE
fix!: Primary keys for `aws_s3_buckets` relations

### PR DESCRIPTION
**Pull request by candiduslynx**
Found during work on https://github.com/cloudquery/cloudquery/issues/13605

### CODE REVIEW

File: plugins/source/aws/resources/services/s3/bucket_cors_rules.go [hunk 1]
```diff
@@ -22,9 +22,10 @@ func bucketCorsRules() *schema.Table {
 		Columns: []schema.Column{
 			client.DefaultAccountIDColumn(false),
 			{
-				Name:     "bucket_arn",
-				Type:     arrow.BinaryTypes.String,
-				Resolver: schema.ParentColumnResolver("arn"),
+				Name:       "bucket_arn",
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/s3/bucket_cors_rules.go:25]: Doesn't this mean we always save a single rule per bucket in the DB? 

Can we have multiple rules per https://docs.aws.amazon.com/AmazonS3/latest/userguide/ManageCorsUsing.html#cors-example-1?

File: plugins/source/aws/resources/services/s3/bucket_lifecycles.go [hunk 2]
```diff
@@ -22,9 +22,10 @@ func bucketLifecycles() *schema.Table {
 		Columns: []schema.Column{
 			client.DefaultAccountIDColumn(false),
 			{
-				Name:     "bucket_arn",
-				Type:     arrow.BinaryTypes.String,
-				Resolver: schema.ParentColumnResolver("arn"),
+				Name:       "bucket_arn",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   schema.ParentColumnResolver("arn"),
+				PrimaryKey: true,
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/s3/bucket_lifecycles.go:28]: We should verify the API for Bucket lifecycle always returns a single item

File: plugins/source/aws/resources/services/s3/bucket_websites.go [hunk 3]
```diff
@@ -21,9 +21,10 @@ func bucketWebsites() *schema.Table {
 		Columns: []schema.Column{
 			client.DefaultAccountIDColumn(false),
 			{
-				Name:     "bucket_arn",
-				Type:     arrow.BinaryTypes.String,
-				Resolver: schema.ParentColumnResolver("arn"),
+				Name:       "bucket_arn",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   schema.ParentColumnResolver("arn"),
+				PrimaryKey: true,
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/s3/bucket_websites.go:27]: We should verify the API for Bucket Website always returns a single item

File: plugins/source/aws/resources/services/s3/bucket_lifecycles.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/source/aws/resources/services/s3/bucket_lifecycles.go:28]: Added `ID` to PK in c2825823e5a4d53ed753f7cc26e2867e8e0ae0d1

File: plugins/source/aws/resources/services/s3/bucket_cors_rules.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/aws/resources/services/s3/bucket_cors_rules.go]: Added `ID` to PK in 6f45f34aff6ef778f06dd815fcbd81247237c429

**Review by erezrokah — APPROVED** [review]

File: plugins/source/aws/resources/services/s3/bucket_cors_rules.go [hunk 4]
```diff
@@ -18,15 +17,8 @@ func bucketCorsRules() *schema.Table {
...
-		Transform:   transformers.TransformWithStruct(&types.CORSRule{}),
-		Columns: []schema.Column{
-			client.DefaultAccountIDColumn(false),
-			{
-				Name:     "bucket_arn",
-				Type:     arrow.BinaryTypes.String,
-				Resolver: schema.ParentColumnResolver("arn"),
-			},
-		},
+		Transform:   transformers.TransformWithStruct(&types.CORSRule{}, transformers.WithPrimaryKeys("ID")),
```
Comment (bbernays) [inline plugins/source/aws/resources/services/s3/bucket_cors_rules.go:20]: We need to verify that all of these fields are required fields.

I just checked this one and it is not required. In [the Examples](https://docs.aws.amazon.com/cli/latest/reference/s3api/get-bucket-cors.html) for the AWS CLI they don't even populate:

```
{
    "CORSRules": [
        {
            "AllowedHeaders": [
                "*"
            ],
            "ExposeHeaders": [
                "x-amz-server-side-encryption"
            ],
            "AllowedMethods": [
                "PUT",
                "POST",
                "DELETE"
            ],
            "MaxAgeSeconds": 3000,
            "AllowedOrigins": [
                "http://www.example.com"
            ]
        },
        {
            "AllowedHeaders": [
                "Authorization"
            ],
            "MaxAgeSeconds": 3000,
            "AllowedMethods": [
                "GET"
            ],
            "AllowedOrigins": [
                "*"
            ]
        }
    ]
}
```

**Review by bbernays — CHANGES_REQUESTED** [review]

File: plugins/source/aws/resources/services/s3/bucket_cors_rules.go (hunk 4 above)
Comment (candiduslynx) [reply plugins/source/aws/resources/services/s3/bucket_cors_rules.go:20]: nice catch!
I'll close this PR for now as noone actually asked to fix those

**Pull request closed without merging**